packages/react/README.md

```diff
@@ -15,3 +15,226 @@
 # @solana/react
 
 This package contains React hooks for building Solana apps.
+
+## Hooks
+
+### `useSignMessage(uiWalletAccount)`
+
+Given a `UiWalletAccount`, this hook returns a function you can call to sign a byte array.
+
+The function accepts a config object with the following properties:
+
+-   `message`: A `Uint8Array` of bytes to sign
+
+That function returns an object with the following properties:
+
+-   `signature`: The 64-byte Ed25519 signature as a `Uint8Array`
+-   `signedMessage`: The `Uint8Array` of bytes signed by the wallet. These bytes might differ from the input bytes if the wallet modified the message
+
+```tsx
+import { useSignMessage } from '@solana/react';
+import React, { useCallback, useState } from 'react';
+
+function SignMessagePanel({ account }) {
+    const signMessage = useSignMessage(account);
+    const [signature, setSignature] = useState(null);
+    const handleSubmit = useCallback(
+        e => {
+            e.preventDefault();
+            const { value: messageText } = e.target.elements.namedItem('message');
+            const message = new TextEncoder().encode(messageText);
+            signMessage({ message }).then(({ signature }) => {
+                setSignature(signature);
+            });
+        },
+        [signMessage],
+    );
+    return (
+        <form onSubmit={handleSubmit}>
+            <textarea name="message"></textarea>
+            {signature ? <span>Signature: {signature.toString()}</span> : null}
+            <button type="submit">Sign Message</button>
+        </form>
+    );
+}
+```
+
+> [!NOTE]
+> There exists a plural version of this hook &ndash; `useSignMessages` &ndash; that accepts multiple inputs and returns an array of outputs.
```

Comment:
Anything else we want to say about this, @jordaaash?

Reply:
I mean if they don't support it, they can throw exceptions, I guess.

One reason that we might have to expose this plural API is that Loris' `@solana/signers` API presumes that everything is plural (ie. `TransactionSendingSigner` et al take in an array of transactions/messages).